### PR TITLE
enhance: refine access log to support format access log by yaml and print name info.

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/28086

### CODE REVIEW

File: internal/proxy/accesslog/formatter.go [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+	return fmt.Sprint(message.XXX_Size())
+}
+
+type BaseResponse interface {
+	GetStatus() *commonpb.Status
+}
+
+func getErrCode(i *GrpcAccessInfo) string {
+	baseResp, ok := i.resp.(BaseResponse)
```
Comment (SimFG) [inline internal/proxy/accesslog/formatter.go:282]: Some response may be `*common.Status`, like `CreateCollection`

File: internal/proxy/authentication_interceptor.go [hunk 2]
```diff
@@ -92,6 +95,7 @@ func AuthenticationInterceptor(ctx context.Context) (context.Context, error) {
 					return nil, status.Errorf(codes.Unauthenticated, "auth check failure, please check username [%s] and password [%s] are correct", username, password)
 				}
 				metrics.UserRPCCounter.WithLabelValues(username).Inc()
+				ctx = context.WithValue(ctx, ContextUsername, username)
```
Comment (SimFG) [inline internal/proxy/authentication_interceptor.go:98]: Same as above

**Comment by aoiasd** [comment]
relate: https://github.com/milvus-io/milvus/issues/28318

File: internal/proxy/accesslog/formatter.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+	}
+	return newFields, newPrefixs
+}
+
+func (f *Formatter) build() {
+	f.prefixs = []string{f.fmt}
+	f.fields = []string{}
+	for mertric := range metricFuncMap {
+		if strings.Contains(f.fmt, mertric) {
+			f.fields, f.prefixs = f.buildMetric(mertric)
```
[the remark this answers is not in the document]
Comment (aoiasd) [reply internal/proxy/accesslog/formatter.go:124]: Prepare when build formatter could help write faster. Use ReplaceAll need Repalce every times when print access log.

File: internal/proxy/accesslog/formatter.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+func NewFormatterManger() *FormatterManger {
+	return &FormatterManger{
+		formatters: make(map[string]*Formatter),
+		methodMap:  make(map[string]string),
+	}
+}
+
+func (m *FormatterManger) Add(name, fmt string) {
```
[the remark this answers is not in the document]
Comment (aoiasd) [reply internal/proxy/accesslog/formatter.go:69]: No now, now we just use this method to init formatter before we use it.

File: internal/proxy/accesslog/info.go [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+func getUserName(i *GrpcAccessInfo) string {
+	username, err := getCurUserFromContext(i.ctx)
+	if err != nil {
+		return unknownString
+	}
+	return username
+}
+
+type SizeResponse interface {
+	XXX_Size() int
```
[the remark this answers is not in the document]
Comment (aoiasd) [reply internal/proxy/accesslog/info.go:165]: Method of go proto Message V2  to get size, proto.size() seems only support photo message v1

File: internal/proxy/accesslog/util.go [hunk 6]
```diff
@@ -23,29 +23,31 @@ import (
...
+	accessInfo := NewGrpcAccessInfo(ctx, info, req)
+	newCtx := context.WithValue(ctx, AccessKey{}, accessInfo)
+	resp, err := handler(newCtx, req)
+	accessInfo.SetResult(resp, err)
+	accessInfo.Write()
 	return resp, err
 }
 
-func Join(path1, path2 string) string {
+func UnaryUpdateAccessInfoInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (aoiasd) [reply internal/proxy/accesslog/util.go:44]: Interceptor was defined by grpc package, now we all interceptor was use interface{}, whether to modify it uniformly.

[the remark this answers is not in the document]
Comment (jiaoew1991) [reply internal/proxy/accesslog/util.go:44]: You can change all of them

[the remark this answers is not in the document]
Comment (aoiasd) [reply internal/proxy/accesslog/util.go]: Changed

**Pull request merged**